### ISSUE TITLE
RuntimeWarning in _xlogx when x has zero values

### ISSUE THREAD

**Issue by guiweber** [raphaelvallat/antropy#10]
In the version currently on github, `_xlogx` uses `numpy.where` to return valid results based on the condition `x==0, 0`. However `numpy.where` still tries to apply the log function to all values of x before trimming the values that meet the condition, resulting in runtime warnings.

To avoid those issues, I would suggest changing the code to something like
```python 
    xlogx = np.zeros_like(x)
    valid = np.nonzero(x)
    xlogx[valid] = x[valid] * np.log(x[valid]) / np.log(base)
    return xlogx
```

As this strictly apply the function to the nonzero elements of x.

If this looks good to you I could submit a PR. Let me know.

**Comment by raphaelvallat** [comment]
Hi @guiweber! cc @jftsang for awareness (see also https://github.com/raphaelvallat/antropy/pull/3). Please feel free to submit a PR!

Thanks,
Raphael

**Comment by guiweber** [comment]
I did some testing and the function I proposed is unfortunately around 1.7 times slower than the old function. However, if a significant portion of x is 0, then the new function is significantly faster (in addition to not spitting RuntimeWarnings). 

For example with x made 50% of zeros, the new function is twice as fast as the old one. Not only does the new function become faster (as expected, since there is less data to compute the log on), but the old function becomes much slower. It seems that either generating the warning or performing log operation on zeros, and then multiplying with -infinity has a very heavy performance cost.

Any thoughts about how we could both avoid the log(0) and keep good performance when there are no or few zeros in x?
